app/javascript/single-file-upload.js

```diff
@@ -0,0 +1,101 @@
+window.addEventListener("DOMContentLoaded", (event) => {
```

Comment:
I'd rather export this inner function and call it than having this run when the file is required. If there's a couple of these on a page it'll add a listener for each one.

Reply:
sorry this comment is unclear for me - couple of what on a page?

Reply:
Components

Reply:
Understand the concerns re: if we had multiple file uploads per page. Currently this javascript is designed for only expecting one file input per page (particularly because there's not really a GDS precedence for forms with multiple single file uploads, so we're not sure if a design would ever look like that) - which is why we use querySelector to get the file input instead of querySelectorAll - this is why I also don't do forEach on the file upload component. forEach does make sense for the saveButtons because we actually expect multiple buttons (Save and continue, Save and come back alter)

Reply:
I will add this explicit information as a comment so that it's clearer